### PR TITLE
Docs-clarify-WireGuard-support

### PR DESCRIPTION
**Pull request by bmckercher123**
## Align OS WireGuard doc with CE

- CE ticket from: https://tigera.atlassian.net/browse/DOCS-409
- Reviewed by Rob for CE: https://github.com/tigera/calico-private/pull/4226
- Note: I spaced it that OS supported WireGuard so failed to start to PR in OS first.

### CODE REVIEW

File: calico/security/encrypt-cluster-pod-traffic.md [hunk 1]
```diff
@@ -19,21 +19,27 @@ This how-to guide uses the following {{site.prodname}} features:
...
+
 - Kubernetes, on-premises
 - EKS using Calico CNI
 - EKS using AWS CNI
 - AKS using Azure CNI
 
-All platforms listed above will encrypt pod-to-pod traffic. Additionally, when using AKS or EKS, host-to-host traffic will also be encrypted, including host-networked pods.
+**Supported traffic for encryption**
+
+- Pod-to-pod traffic
```
Comment (caseydavenport) [inline calico/security/encrypt-cluster-pod-traffic.md:31]: Pod-to-pod traffic is misleading and potentially wrong.

For example, we don't encrypt pod -> pod if it's on the same host - we only encrypt the traffic on the host-to-host network hop.

I'd probably say that we support "host-to-host encryption for pod traffic, and optionally for direct node to node communication as well"

File: calico/security/encrypt-cluster-pod-traffic.md [hunk 2]
```diff
@@ -19,21 +19,27 @@ This how-to guide uses the following {{site.prodname}} features:
...
+
+  > **Note**: Some node operating systems do not support Wireguard, or do not have it installed by default. Enabling {{site.prodname}} Wireguard encryption does not require all nodes to be installed with Wireguard. However, traffic to or from a node that does not have Wireguard installed, will not be encrypted.
+  {: .alert .alert-info}
+
+- IP addresses for every node in the cluster. This is required to establish secure tunnels between the nodes. {{site.prodname}} can automatically do this using [IP Setting]({{site.baseurl}}/reference/node/configuration#ip-setting) and [IP autodetection methods]({{site.baseurl}}/reference/node/configuration#ip-autodetection-methods) available under [calico/node]({{site.baseurl}}/reference/node/configuration) resource.
+    - Under [installation]({{site.baseurl}}/reference/installation/api), set the [autodetection method]({{site.baseurl}}/reference/installation/api#operator.tigera.io/v1.NodeAddressAutodetection) (`nodeAddressAutodetectionV4` and/or `nodeAddressAutodetectionV6`) for your cluster.
```
Comment (caseydavenport) [inline calico/security/encrypt-cluster-pod-traffic.md:42]: In OSS we support both operator and manifest based installation methods, and this bullet point only applies to operator.

I think this sub bullet is actually just redundant and that the first bullet covers what we need.

File: calico/security/encrypt-cluster-pod-traffic.md [hunk 3]
```diff
@@ -141,9 +147,6 @@ To install WireGuard for OpenShift v4.8:
 
 #### Enable WireGuard for a cluster
 
-   > **Note**: Nodes that do not support WireGuard will not be secured by WireGuard tunnels, even if traffic running on the node to and from the pods goes to nodes that do support WireGuard.
```
Comment (caseydavenport) [inline calico/security/encrypt-cluster-pod-traffic.md:144]: Why remove this note?

Comment (bmckercher123) [reply calico/security/encrypt-cluster-pod-traffic.md:144]: This was the text that Support said was confusing. The content was moved to Before  you begin so doesn't need to be repeated here.

File: calico/security/encrypt-cluster-pod-traffic.md (hunk 2 above)
Comment (bmckercher123) [reply calico/security/encrypt-cluster-pod-traffic.md:42]: Removed second bullet.

File: calico/security/encrypt-cluster-pod-traffic.md (hunk 1 above)
Comment (bmckercher123) [reply calico/security/encrypt-cluster-pod-traffic.md]: Updated using your terms, and also updated Warnings to use same.

**Review by caseydavenport — APPROVED** [review]

**Pull request merged**